pkg/prometheus/promcfg.go

```diff
@@ -2149,13 +2149,10 @@ func (cg *ConfigGenerator) generateAlertmanagerConfig(alerting *monitoringv1.Ale
 
 		cfg = cg.WithMinimumVersion("2.48.0").addSigv4ToYaml(cfg, fmt.Sprintf("alertmanager/auth/%d", i), store, am.Sigv4)
 
-		apiVersionCg := cg.WithMinimumVersion("2.11.0")
-		switch am.APIVersion {
-		case "v1":
-			// API v1 isn't supported anymore by Prometheus v3.
-			cfg = apiVersionCg.WithMaximumVersion("2.999.0").AppendMapItem(cfg, "api_version", am.APIVersion)
-		case "v2":
-			cfg = apiVersionCg.AppendMapItem(cfg, "api_version", am.APIVersion)
+		// API v1 isn't supported anymore by Prometheus v3.
+		if (am.APIVersion.ToUpper() == monitoringv1.AlertmanagerAPIVersion1 && cg.version.Major <= 2) ||
+			am.APIVersion.ToUpper() == monitoringv1.AlertmanagerAPIVersion2 {
+			cfg = cg.WithMinimumVersion("2.11.0").AppendMapItem(cfg, "api_version", am.APIVersion)
```

Comment:
(suggestion) I would keep the switch/case.

Reply:
Thanks @simonpasquier 
Keep it  switch/case